polar-core/src/vm.rs

```diff
@@ -1172,42 +1172,54 @@ impl PolarVirtualMachine {
 
                 let simplified = simplified.value().as_expression()?;
 
+                use Operator::*;
                 // TODO (dhatch): what if there is more than one var = dot_op constraint?
                 // What if the one there is is in a not, or an or, or something
-                let lhs_of_matches = simplified
+                let mut lhss_of_matches: Vec<_> = simplified
                     .constraints()
                     .into_iter()
-                    .find_map(|c| {
+                    .filter_map(|Operation { operator, args }| match operator {
```

Comment:
The main change here is to use `filter_map` instead of `find_map`. `choose_conditional` takes a collection as its first arg anyway so it's compatible